### ISSUE TITLE
how to implement the api in pytorch with tensorflow 2.0

### ISSUE THREAD

**Issue by HuangZhanPeng** [tensorflow/tensorflow#27642]
Once I have ask question following:
how to implement the api in pytorch with tensorflow 2.0There is two api in pytorch, one is "nn.utils.rnn.pack_padded_sequence", other is "pad_packed_sequence", I want to ask is there any api or implement to achieve the similar function of pytorch in tensorflow 2.0, thank you~~

And @ebrevdo give me the solution:
If you call ` tf.nn.{dynamic_,}rnn` with `sequence_length` vector provided and use padded inputs, it will behave the same as if you used pytorch's pack_padded_sequence. Specifically, the final state will properly contain the state of the final step for each batch, and the output values in a batch row b corresponding to time points after `sequence_length[b]` will be zeroed out.

but in TensorFlow 2.0, now there is no `tf.nn.{dynamic_,}rnn`, how to deal with variable length?

**Comment by ebrevdo** [comment]
It's there, just in a different namespace.  I think in tf.rnn.  I believe
keras lstm also provides similar functionally with mask arguments.

**Comment by mihaimaruseac** [comment]
I think this should go to StackOverflow as it is not related to a bug or feature request in TensorFlow.

**Comment by HuangZhanPeng** [comment]
@mihaimaruseac I think that is a huge api change, but tf 2.0 did not provide analogous api, I still insist that may be a feature request in TensorFlow

**Comment by HuangZhanPeng** [comment]
@ebrevdo [#26974](https://github.com/tensorflow/tensorflow/issues/26974), Yes, there is similar function to achieve that, Although this does not really handle the variable sequence length. And in tf 2.0 may be there is not `tf.rnn`.

May be in the Effective TensorFlow 2.0 tutorial, it can be soluted with the code sample:

```
class DynamicRNN(tf.keras.Model):

  def __init__(self, rnn_cell):
    super(DynamicRNN, self).__init__(self)
    self.cell = rnn_cell

  def call(self, input_data):
    # [batch, time, features] -> [time, batch, features]
    input_data = tf.transpose(input_data, [1, 0, 2])
    outputs = tf.TensorArray(tf.float32, input_data.shape[0])
    state = self.cell.zero_state(input_data.shape[1], dtype=tf.float32)
    for i in tf.range(input_data.shape[0]):
      output, state = self.cell(input_data[i], state)
      outputs = outputs.write(i, output)
    return tf.transpose(outputs.stack(), [1, 0, 2]), state
```

**Comment by mihaimaruseac** [comment]
Side note: why did you edit the title and descriptions? You know GitHub stores all edits and anyone can see them, right? It just causes hassles when trying to get back the context
